### PR TITLE
Task 6 get comments

### PR DESCRIPTION
**Pull request by roseannac21**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -102,5 +94,30 @@ describe("app tests", () => {
...
+      test("status 200", () => {
+        return request(app).get("/api/reviews/3/comments/").expect(200);
+      })
+      test("returns the array of comments from the specified review_id", () => {
+        return request(app).get("/api/reviews/3/comments/").expect(200).then((response) => {
+          const commentsForReview = response.body;
+          expect(commentsForReview).toHaveLength(3);
+          commentsForReview.forEach((obj) => {
+            expect(obj.review_id).toBe(3);
+          })
```
Comment (alexdevdavis) [inline __tests__/app.test.js:108]: this is a neat test 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -102,5 +94,30 @@ describe("app tests", () => {
...
+      test("error handling- 400- invalid data type for review_id", () => {
+        return request(app).get("/api/reviews/hello/comments/").expect(400).then(({body}) => {
+          expect(body.msg).toBe("invalid data type")
+        })
+      })
+      test("error handling- 404- id number doesn't exist but is valid data type", () => {
+        return request(app).get("/api/reviews/99999/comments/").expect(404).then(({body}) => {
+          expect(body.msg).toBe("path not found");
+        })
+      })
```
Comment (alexdevdavis) [inline __tests__/app.test.js:120]: good to see 404 and 400 here

File: db/app.js [hunk 3]
```diff
@@ -15,6 +15,8 @@ app.get("/api/reviews/", getReviews);
 
 app.get("/api/reviews/:review_id/", getReview);
 
+app.get("/api/reviews/:review_id/comments/", getCommentsForReview);
+
 app.use((req, res, next) => {
   res.status(404).send({msg: "path not found"})
 })
```
Comment (alexdevdavis) [inline db/app.js:22]: have a look at how you might create a generic custom error handler that checks if there's err.status && err.msg, and will send whatever that status and message is. That way you can define status and msg when you reject a promise.

File: db/app.js [hunk 4]
```diff
@@ -27,4 +29,10 @@ app.use((err, req, res, next) => {
     }
   });
 
+  app.use((err, req, res, next) => {
+    if (err) {
+      console.log(err);
```
Comment (alexdevdavis) [inline db/app.js:34]: this error handler will just hand in present state if you get some kind of server error, fyi

File: db/controller.js [hunk 5]
```diff
@@ -14,21 +15,21 @@ const getReviews = (request, response, next) => {
...
 };
 
 const getCommentsForReview = (request, response, next) => {
-
   const reviewId = request.params.review_id;
-
-  fetchCommentsForReview(reviewId).then((comments) => {
-    response.status(200).send({ comments });
+  
+Promise.all([fetchCommentsForReview(reviewId), fetchReviewById(reviewId)]).then((comments) => {
```
Comment (alexdevdavis) [inline db/controller.js:23]: nicely done 👍

File: db/controller.js [hunk 6]
```diff
@@ -14,21 +15,21 @@ const getReviews = (request, response, next) => {
...
+    if (comments[0].length === 0 && comments[1].length === 0) {
+      next();
```
Comment (alexdevdavis) [inline db/controller.js:25]: further to custom error suggestion, you could Promise.reject and define the status and msg here, instead of invoking a generic next with no argument

File: db/model.js [hunk 7]
```diff
@@ -26,8 +26,15 @@ const fetchReviewById = (id) => {
 
     return db.query(queryStr).then(({rows}) => {
         return rows;
+    })
+}
+
+const fetchCommentsForReview = (id) => {
+    let queryStr = `SELECT * FROM comments WHERE review_id = ${id};`
```
Comment (alexdevdavis) [inline db/model.js:33]: use dollar syntax and values array to help protect from SQL injection here

File: __tests__/app.test.js [hunk 8]
```diff
@@ -102,5 +94,30 @@ describe("app tests", () => {
         })
       })
     })
+  
+    describe("task 4 get comments for specified review", () => {
```
Comment (alexdevdavis) [inline __tests__/app.test.js:98]: ideally we'd also test to ensure that we get an empty array when request is valid, but review has no comments yet

**Review by alexdevdavis — APPROVED** [review]
Nicely done here, Roseanna. 👍 
- I've given you some pointers about how to have more control over your error handling
- I don't think app, controller and model files should live in db folder.
